File: C_plus/Longest Common Prefix.cpp
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        string result;
        
        if(strs.empty()) return result;
        
        for(int i = 0; i< strs[0].size();i++) {
            
            char temp = strs[0][i];
            
            for(int j = 1;j<strs.size();j++) {
                
                if(strs[j][i] != temp||i > strs[j].size()) {
                    
                    return result;
                    
                }
                
            }
            
            result.push_back(temp);
            
        }
        
        return result;
        
    }

File: python/Longest Common Prefix.py
def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        result = ''
        
        if len(strs) == 0 :
            return ''
            
        for i in range(len(strs[0])):
            temp = strs[0][i]
            for j in range(1,len(strs)):
                if i+1> len(strs[j]) or temp != strs[j][i]:
                    return result
            result += temp
        return result
